things/service/src/main/java/org/eclipse/ditto/things/service/persistence/actors/ThingSupervisorActor.java

```diff
@@ -315,6 +329,36 @@ protected CompletionStage<Object> modifyTargetActorCommandResponse(final Signal<
                 .run(materializer);
     }
 
+    @Override
+    protected CompletableFuture<Object> handleTargetActorException(final Throwable error, final Signal<?> enforcedSignal) {
+        if (enforcedSignal instanceof CreateThing createThing) {
+            CompletableFuture<Object> responseFuture = new CompletableFuture<>();
+            getSelf().tell(RollbackCreatedPolicy.of(createThing, error, responseFuture), getSelf());
+            return responseFuture;
+        }
+        return CompletableFuture.failedFuture(error);
+    }
+
+    private void handlerRollbackCreatedPolicy(final RollbackCreatedPolicy rollback) {
+        if (policyCreatedEvent != null) {
```

Comment:
After the if/else the field `policyCreatedEvent` must be set to null again. Otherwise it will always stay non null